libs/vgc/ui/menubutton.h

```diff
@@ -0,0 +1,246 @@
+///
+/// It destroys itself whenever its action gets destroyed.
+///
+class VGC_UI_API ActionButton : public Flex {
+private:
+    VGC_OBJECT(ActionButton, Flex)
+
+protected:
+    ActionButton(Action* action, FlexDirection layoutDirection);
+
+public:
+    /// Creates an MenuButton with the given `action`.
+    ///
+    static ActionButtonPtr
+    create(Action* action, FlexDirection layoutDirection = FlexDirection::Column);
+
+    Action* action() const {
+        return action_;
+    }
+
+    bool isDisabled() const {
+        return action_ ? action_->isDisabled() : true;
+    }
+
+    bool isMenu() const {
+        return action_ ? action_->isMenu() : false;
+    }
+
+    bool click() {
+        return tryClick_();
+    }
+
+    /// This signal is emitted when the button is clicked and is not disabled.
+    ///
+    VGC_SIGNAL(clicked);
+
+protected:
+    // Reimplementation of Object virtual methods
+    void onDestroyed() override;
+
+    // Reimplementation of Widget virtual methods
+    void onWidgetRemoved(Widget* child) override;
+    bool onMouseEnter() override;
+    bool onMouseLeave() override;
+    bool onMousePress(MouseEvent* event) override;
+    void onResize() override;
+    void onStyleChanged() override;
+    void onPaintCreate(graphics::Engine* engine) override;
+    void onPaintDraw(graphics::Engine* engine, PaintOptions options) override;
+    void onPaintDestroy(graphics::Engine* engine) override;
+
+    virtual void onClicked();
+
+    Widget* iconWidget() const {
+        return iconWidget_;
+    }
+
+    Label* textLabel() const {
+        return textLabel_;
+    }
+
+    Label* shortcutLabel() const {
+        return shortcutLabel_;
+    }
+
+    void setActive(bool isActive);
+
+private:
+    // Components
+    Action* action_ = nullptr;
+    Widget* iconWidget_ = nullptr;
+    Label* textLabel_ = nullptr;
+    Label* shortcutLabel_ = nullptr;
+
+    // Style
+    bool isActive_ = false;
+
+    // Background
+    graphics::GeometryViewPtr triangles_;
+    bool reload_ = true;
+
+    // Behavior
+    bool tryClick_();
+    void onActionChanged_();
+
+    VGC_SLOT(onActionChangedSlot_, onActionChanged_);
+    VGC_SLOT(onActionAboutToBeDestroyed_, destroy);
+};
+
+/// \enum vgc::ui::MenuDropDirection
+/// \brief The direction in which a dropdown menu should appear.
+///
+enum class MenuDropDirection {
+    Horizontal,
+    Vertical,
+};
+
+/// \class vgc::ui::MenuButton
+/// \brief A button with a special layout for Menus.
+///
+class VGC_UI_API MenuButton : public ActionButton {
+private:
+    VGC_OBJECT(MenuButton, ActionButton)
+    friend Menu;
+
+protected:
+    MenuButton(Action* action, FlexDirection layoutDirection);
+
+public:
+    /// Creates an MenuButton with the given `action`.
+    ///
+    static MenuButtonPtr
+    create(Action* action, FlexDirection layoutDirection = FlexDirection::Column);
+
+    MenuDropDirection menuDropDir() const {
```

Comment:
`menuDropDirection()` and `menuDropDirection_`. OK to keep using "dropDir" local variables in .cpp files though.